From f52823719922c48e050e1235e0bfb0649e04de79 Mon Sep 17 00:00:00 2001
From: Subhendu Sekhar Behera <sbehera@broadcom.com>
Date: Sun, 2 Mar 2014 18:38:47 +0530
Subject: general: Handle debug messages.

set NETSOC_APIDBG_TRACE to 0x10 and get nae_debug_level in "netsoc_lib_param"
[Based on SDK 3.2]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
Signed-off-by: Nam Ninh <nam.ninh@windriver.com>

diff --git a/arch/mips/netlogic/lib/netlib/include/netsoc_libiface.h b/arch/mips/netlogic/lib/netlib/include/netsoc_libiface.h
index 22ce6f6..f115794 100644
--- a/arch/mips/netlogic/lib/netlib/include/netsoc_libiface.h
+++ b/arch/mips/netlogic/lib/netlib/include/netsoc_libiface.h
@@ -57,7 +57,7 @@ enum netsoc_api_dbglevels{
 	NETSOC_APIDBG_CONFIG = 0x4,
 	NETSOC_APIDBG_PORT = 0x8,
 	NETSOC_APIDBG_GLOBAL = NETSOC_APIDBG_PORT | NETSOC_APIDBG_CONFIG,
-	NETSOC_APIDBG_TRACE = 0xFF,
+	NETSOC_APIDBG_TRACE = 0x10,
 };
 
 
@@ -133,6 +133,7 @@ struct netsoc_lib_param {
         void *(*phys_to_virt)(uint64_t);
 	/*library params */
 	int kr_en;
+	int netsoc_debug_level;
 };
 
 static inline void *netsoc_api_contig_malloc(uint32_t align, uint32_t size)
diff --git a/arch/mips/netlogic/lib/netlib/src/netsoc_api.c b/arch/mips/netlogic/lib/netlib/src/netsoc_api.c
index ce3c419..cbc026d 100644
--- a/arch/mips/netlogic/lib/netlib/src/netsoc_api.c
+++ b/arch/mips/netlogic/lib/netlib/src/netsoc_api.c
@@ -1759,6 +1759,7 @@ int  brcm_netsoc_lib_init(struct netsoc_lib_param *libparam)
                 netsoc_lib_contig_free = libparam->contig_free;
                 netsoc_lib_phystovirt = libparam->phys_to_virt;
 		netsoc_lib_virttophys = libparam->virt_to_phys;
+		netsoc_api_dbglvl = libparam->netsoc_debug_level;
 	}     
 
 	if(__netsoc_validate_lib_param(libparam)<0){
-- 
1.7.1

